inc/Engine/License/Upgrade.php

```diff
@@ -282,6 +282,10 @@ private function can_use_promo() {
 			return false;
 		}
 
+		if ( ( 14 * DAY_IN_SECONDS ) > time() - $this->user->get_creation_date() ) {
```

Comment:
u can create a private method to handle this condition, u may call it `is_new_user`, so we bailout here if the user is a new user (registered for less than 14 days)